### PR TITLE
Drafty handling refactored to use trees

### PR DESCRIPTION
**Pull request by or-else**
Drafty formatting refactored to use trees: a doc is converted to a tree where each node is uniformly formatted, like

[_quick **fox** jumps_ over] ->
```
[[EM: quick, [ST: fox], jumps], over]
```
It simplified removal of quoted text, shortening etc. Also added a few more tests for various corner cases.

--

There are three unrelated changes:
* `auth_basic.go`: added '**.**' as a permitted character to logins: https://groups.google.com/g/tinode/c/Ncl0MG-QzQM/m/cvi88V0ACAAJ
* Added a comment to mysql `schema.sql` as a reaction to this: https://groups.google.com/g/tinode/c/meEqSq6XzKU/m/tzGrao6-CAAJ
* Fixed a bug in `store.go` related to handling of out-of-band uploaded files.

### CODE REVIEW

File: server/db/mysql/schema.sql [hunk 1]
```diff
@@ -1,3 +1,13 @@
+# THIS SCHEMA FILE IS FOR REFERENCE/DOCUMENTATION ONLY!
+# DO NOT USE IT TO INITIALIZE THE DATABASE.
+# Read installation instructions first.
+
+# The following line will produce an intentional error.
+
+'READ INSTALLATION INSTRUCTIONS!';
```
Comment (aforge) [inline server/db/mysql/schema.sql:7]: Perhaps, add a link here?

**Review by aforge — COMMENTED** [review]
How hard would it be to add a test for `drafty.go`?

**Comment by or-else** [comment]
> How hard would it be to add a test for drafty.go?

Which test do you mean? There are a few unit tests there.

File: server/db/mysql/schema.sql (hunk 1 above)
Comment (or-else) [reply server/db/mysql/schema.sql:7]: I thought about it and decided against it: we will forget about it and it will get stale. It's just a forcing function to make people search for answers.

File: server/drafty/drafty.go [hunk 2]
```diff
@@ -22,368 +47,572 @@ type span struct {
...
-	"CO": {"", false},
+	"CO": {"`", false},
 	"BR": {"\n", true},
 	"LN": {"", false},
 	"MN": {"", false},
 	"HT": {"", false},
-	"IM": {"", true},
+	"IM": {"", false},
 	"EX": {"", true},
+	"QQ": {"", false},
```
Comment (aforge) [inline server/drafty/drafty.go:168]: Can you add `IC` here as well?

**Comment by aforge** [comment]
> > How hard would it be to add a test for drafty.go?
> 
> Which test do you mean? There are a few unit tests there.

My bad. Sorry for confusion.

**Review by aforge — APPROVED** [review]

File: server/drafty/drafty.go (hunk 2 above)
Comment (or-else) [reply server/drafty/drafty.go]: Actually, only non-blank tags are needed here. I added `IC` handling further down in the code + test.
